### PR TITLE
Ingest an updated version of atom-renderer for audio atoms

### PR DESCRIPTION
**Pull request by JustinPinner**
## What does this change?
We're getting ready to bring in an updated build of the atom-renderer library for audio atoms. This is still WIP until that's ready for PROD consumption - i.e. when [this PR](https://github.com/guardian/atom-renderer/pull/58) is all done and dusted.

This change also shifts responsibility for providing ad-free or Acast-wrapped source URLs over to the atom-renderer, but dotcom needs to pass the ad-free user status and the Acast switch value for that to work.

## Screenshots
(captured locally) - the delay before initial playback (00:07:07 in the image below) is a known issue

audio-atom-with-extra-scrubbability

## What is the value of this and can you measure success?
Helps to unify the app and web views of audio atoms

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)
- [x] No

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/master/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)
- [x] Accessibility tests are still pending for this feature

### Tested

- [x] Locally
- [x] On CODE (optional)

```
Browser		Mobile		Desktop
========================================
Chrome		pass**		pass
Firefox		pass**		pass
Safari		pass*		pass
Edge (18)		        pass***i
IE (11)			        fail***ii		

* iOS 11
** Android 5
*** Win10

i Slightly odd rendering of scrubber control
ii Initial setup broken, playback & scrubbing works, just looks terrible
```

### CODE REVIEW

File: common/app/views/support/HtmlCleaner.scala [hunk 1]
```diff
@@ -697,12 +699,21 @@ case class AtomsCleaner(
             atomContainer.attr("data-atom-id", atomId)
             atomContainer.attr("data-atom-type", atomType)
 
-            val html = if(atomData.isInstanceOf[AudioAtom] && isAdFree(request)) {
-              views.html.fragments.atoms.atom(atomData, Atoms.articleConfig, shouldFence, amp, mediaWrapper, posterImageOverride).toString().replaceAll("flex.acast.com/", "")
+            val articleConfig: ArticleConfiguration = if (atomData.isInstanceOf[AudioAtom]) {
```
Comment (regiskuckaertz) [inline common/app/views/support/HtmlCleaner.scala]: Hhm, ideally we build one article configuration valid for all atoms.

File: common/app/model/content/Atoms.scala [hunk 2]
```diff
@@ -228,10 +228,15 @@ final case class AudioAtom(
 
 object Atoms extends common.Logging {
 
-  def articleConfig = ArticleConfiguration(
-    ajaxUrl = Configuration.ajax.url,
-    commonsdivisionConfiguration = ArticleConfiguration.CommonsdivisionConfiguration(showMps = true)
-  )
+  def articleConfig(isAdFree: Boolean = false, useAcast: Boolean = false) = {
```
Comment (regiskuckaertz) [inline common/app/model/content/Atoms.scala]: How about
```scala
def articleConfig(request: Request) = ArticleConfiguration(
  ajaxUrl = Configuration.ajax.url,
  audio = AudioConfiguration(!isAdFree(request) && Switches.Acast.isSwitchedOn),
  commonsdivisionConfiguration = CommonsdivisionConfiguration(true)
)
```

File: common/app/views/support/HtmlCleaner.scala (hunk 1 above)
Comment (JustinPinner) [reply common/app/views/support/HtmlCleaner.scala]: I think we've addressed this now. Will mark as resolved.

**Review by regiskuckaertz — APPROVED** [review]
Yay

**Pull request merged**